runtime/src/accounts_hash.rs

```diff
@@ -667,16 +664,13 @@ impl AccountsHash {
                 // reverse this list because get_item can remove first_items[*i] when *i is exhausted
                 //  and that would mess up subsequent *i values
                 duplicate_pubkey_indexes.iter().rev().for_each(|i| {
-                    let (exhausted, _) = Self::get_item(
+                    Self::get_item(
                         *i,
                         pubkey_bin,
                         &mut first_items,
                         pubkey_division,
                         &mut indexes,
                     );
-                    if exhausted {
-                        min_index -= 1;
-                    }
```

Comment:
I assume this was removed because it's not needed. What's the observation that enabled this?